RxCocoa/iOS/UITableView+Rx.swift

```diff
@@ -46,16 +48,18 @@ extension UITableView {
     public func rx_itemsWithCellIdentifier<S: SequenceType, Cell: UITableViewCell, O : ObservableType where O.E == S>
         (cellIdentifier: String, cellType: Cell.Type = Cell.self)
         (source: O)
-        (configureCell: (Int, S.Generator.Element, Cell) -> Void)
+        -> (configureCell: (Int, S.Generator.Element, Cell) -> Void)
         -> Disposable {
-        let dataSource = RxTableViewReactiveArrayDataSourceSequenceWrapper<S> { (tv, i, item) in
-            let indexPath = NSIndexPath(forItem: i, inSection: 0)
-            let cell = tv.dequeueReusableCellWithIdentifier(cellIdentifier, forIndexPath: indexPath) as! Cell
-            configureCell(i, item, cell)
-            return cell
+        return { configureCell in
+            let dataSource = RxTableViewReactiveArrayDataSourceSequenceWrapper<S> { (tv, i, item) in
+                let indexPath = NSIndexPath(forItem: i, inSection: 0)
+                let cell = tv.dequeueReusableCellWithIdentifier(cellIdentifier, forIndexPath: indexPath) as! Cell
+                configureCell(i, item, cell)
+                return cell
+            }
+                
+            return self.rx_itemsWithDataSource(dataSource)(source: source)
```

Comment:
I'm also wondering if we should use `[weak self]`, because of currying (since it just function which returns closure)